linux-release does not build on Ubuntu 22.04

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

```
FAILED: source/CMakeFiles/EndlessSkyLib.dir/Release/GameWindow.cpp.o 
/usr/bin/c++ -DCMAKE_INTDIR=\"Release\" -isystem /usr/include/SDL2 -O3 -DNDEBUG -std=c++17 -flto=auto -fno-fat-lto-objects -fdiagnostics-color=always -Wall -pedantic-errors -Wold-style-cast -fno-rtti -MD -MT source/CMakeFiles/EndlessSkyLib.dir/Release/GameWindow.cpp.o -MF source/CMakeFiles/EndlessSkyLib.dir/Release/GameWindow.cpp.o.d -o source/CMakeFiles/EndlessSkyLib.dir/Release/GameWindow.cpp.o -c /home/redacted/src/sky/cmake-text-wall/source/GameWindow.cpp
/home/redacted/src/sky/cmake-text-wall/source/GameWindow.cpp: In static member function ‘static bool GameWindow::Init(bool)’:
/home/redacted/src/sky/cmake-text-wall/source/GameWindow.cpp:85:29: error: ‘SDL_HINT_VIDEODRIVER’ was not declared in this scope; did you mean ‘SDL_HINT_RENDER_DRIVER’?
   85 |                 SDL_SetHint(SDL_HINT_VIDEODRIVER, "dummy");
      |                             ^~~~~~~~~~~~~~~~~~~~
      |                             SDL_HINT_RENDER_DRIVER
```

### Steps to Reproduce

1. Get a Linux machine with Ubuntu 22.04
2. Install the packages listed in `doc/readme-cmake.md`
3. cmake --preset linux
4. cmake --build --verbose --preset linux-release

### Expected Behavior

It compiles.

### Screenshots

_No response_

### Link to save file

_No response_

### Operating System

`Linux 6.2.0-39-generic #40~22.04.1-Ubuntu x86_64 GNU/Linux`

### Game Source

GitHub Releases

### Game Version

b3a8d95ac

### Additional Information

Problem started with this hash:

4be1d8c66173fa980f3874819bd6e615576448de

which added the problematic line.

```c++
SDL_SetHint(SDL_HINT_VIDEODRIVER, "dummy");
```

--- a/source/GameWindow.cpp
+++ b/source/GameWindow.cpp
@@ -82,7 +82,11 @@ bool GameWindow::Init(bool headless)
 
 	// When running the integration tests, don't create a window nor an OpenGL context.
 	if(headless)
+#if defined(__linux__) && !SDL_VERSION_ATLEAST(2, 0, 22)
+		setenv("SDL_VIDEODRIVER", "dummy");
+#else
 		SDL_SetHint(SDL_HINT_VIDEODRIVER, "dummy");
+#endif
 
 	// This needs to be called before any other SDL commands.
 	if(SDL_Init(SDL_INIT_VIDEO) != 0)

--- a/source/GameWindow.cpp
+++ b/source/GameWindow.cpp
@@ -83,7 +83,7 @@ bool GameWindow::Init(bool headless)
 	// When running the integration tests, don't create a window nor an OpenGL context.
 	if(headless)
 #if defined(__linux__) && !SDL_VERSION_ATLEAST(2, 0, 22)
-		setenv("SDL_VIDEODRIVER", "dummy");
+		setenv("SDL_VIDEODRIVER", "dummy", true);
 #else
 		SDL_SetHint(SDL_HINT_VIDEODRIVER, "dummy");
 #endif

